src/quick-lint-js/error.h

```diff
@@ -789,6 +789,13 @@
       ERROR(QLJS_TRANSLATABLE("missing parameters for arrow function"),        \
             arrow))                                                            \
                                                                                \
+  QLJS_ERROR_TYPE(                                                             \
+      error_missing_parens_self_invoked_function, "E0211",                     \
+      { source_code_span func; },                                              \
```

Comment:
If the error spans the entire function, that could mean it spans hundreds of lines. Huge error spans lead to a bad user experience.

I think we should report an error like this instead:
```
    () => {}()
            ^ missing parentheses around self-invoked function
    ^ function begins here
```
This would mean storing two empty `source_code_span`s. Kinda dumb, but oh well.

(The CLI error reporting doesn't communicate in this way currently. You need to use an IDE integration to see the spans.)